skypy/utils/tests/test_random.py

```diff
@@ -1,52 +1,58 @@
-import scipy.integrate
-from skypy.utils import random, special
 
+from scipy.stats import kstest
 
-def test_schechter_cdf():
-    # Test the _schechter_cdf function against precomputed values
-    x = np.logspace(np.log10(0.1), np.log10(10), 10)
-    cdf = random._schechter_cdf(x, min(x), max(x), -1.3)
-    sample = np.array([0., 0.31938431, 0.57066007, 0.75733831, 0.88345969,
-                       0.95631112, 0.98886695, 0.99846589, 0.99992254, 1.])
 
-    np.testing.assert_allclose(cdf, sample)
+def test_schechter():
 
+    from skypy.utils.random import schechter
 
-def test_schechter():
+    def schechter_cdf_gen(alpha, x_min, x_max, resolution=100000):
+        lnx = np.linspace(np.log(x_min), np.log(x_max), resolution)
+
+        cdf = np.exp((alpha + 1)*lnx - np.exp(lnx))
+        np.cumsum((cdf[:-1] + cdf[1:])/2*np.diff(lnx), out=cdf[1:])
+        cdf[0] = 0
+        cdf /= cdf[-1]
+
+        return lambda x: np.interp(np.log(x), lnx, cdf)
 
     # Test the schechter function, sampling dimensionless x values
     alpha = -1.3
     x_min = 1e-10
     x_max = 1e2
 
-    def calc_cdf(x):
-        a = special.gammaincc(alpha + 1, x_min)
-        b = special.gammaincc(alpha + 1, x)
-        c = special.gammaincc(alpha + 1, x_min)
-        d = special.gammaincc(alpha + 1, x_max)
-        return (a - b) / (c - d)
+    x = schechter(alpha, x_min=x_min, x_max=x_max, size=1000)
 
-    sample = random.schechter(alpha, x_min=x_min, x_max=x_max,
-                              resolution=100, size=1000)
+    assert np.all(x >= x_min)
+    assert np.all(x <= x_max)
 
     # Test the distribution of galaxy properties follows the right distribution
-    p_value = scipy.stats.kstest(sample, calc_cdf)[1]
-    assert p_value > 0.01
+    cdf = schechter_cdf_gen(alpha, x_min, x_max)
+    _, p = kstest(x, cdf)
+    assert p > 0.01
 
     # Test output shape when scale is a scalar
     scale = 5
-    samples = random.schechter(alpha, x_min, x_max, scale=scale)
+    samples = schechter(alpha, x_min, x_max, scale=scale)
     assert np.shape(samples) == np.shape(scale)
 
     # Test output shape when scale is an array
     scale = np.random.uniform(size=10)
-    samples = random.schechter(alpha, x_min, x_max, scale=scale)
+    samples = schechter(alpha, x_min, x_max, scale=scale)
     assert np.shape(samples) == np.shape(scale)
 
-    # When alpha = 0 check the samples match an exponential distribution
-    alpha, scale, size = 0, 5, 1000
-    samples = random.schechter(alpha, x_min, x_max, size=size, scale=scale)
-    p_value = scipy.stats.kstest(samples, 'expon', args=(0, scale))[1]
-    assert p_value > 0.01
+
+def test_schechter_gamma():
+
+    from skypy.utils.random import schechter
+
+    # when alpha > -1, x_min ≈ 0, x_max ≈ ∞, distribution is gamma
+    alpha = np.random.uniform(-1, 2)
+    x_min = 1e-10
+    x_max = 1e+10
+
+    x = schechter(alpha, x_min, x_max, resolution=100000, size=1000)
+
+    _, p = kstest(x, 'gamma', args=(alpha+1,))
```

Comment:
```suggestion
    scale = 5
    x = schechter(alpha, x_min, x_max, resolution=100000, size=1000, scale=scale)

    _, p = kstest(x, 'gamma', args=(alpha+1, 0, scale))
```
I understand switching from the old 'expon' to 'gamma' but can we still test `scale` here as well? I think this is correct but perhaps check it